packages/client/tests/functional/_utils/setupTestSuiteMatrix.ts

```diff
@@ -44,22 +44,20 @@ function setupTestSuiteMatrix(
   const originalEnv = process.env
   const suiteMeta = getTestSuiteMeta()
   const suiteConfig = getTestSuiteConfigs(suiteMeta)
-  const suiteTable = getTestSuiteTable(suiteMeta, suiteConfig)
-  const forceInlineSnapshot = process.argv.includes('-u')
+  const testPlan = getTestSuitePlan(suiteMeta, suiteConfig)
   checkMissingProviders({
     suiteConfig,
     suiteMeta,
     options,
   })
-  ;(forceInlineSnapshot ? [suiteTable[0]] : suiteTable).forEach((suiteEntry) => {
-    const [suiteName, suiteConfig] = suiteEntry
-
+  for (const { name, suiteConfig, skip } of testPlan) {
+    const describeFn = skip ? describe.skip : describe
     // we don't run tests for some providers that we want to skip on the CI
-    if (suiteConfig['provider']?.toLowerCase() === 'mongodb' && process.env.TEST_SKIP_MONGODB) return
-    if (suiteConfig['provider']?.toLowerCase() === 'sqlserver' && process.env.TEST_SKIP_MSSQL) return
-    if (suiteConfig['provider']?.toLowerCase() === 'cockroachdb' && process.env.TEST_SKIP_COCKROACHDB) return
+    if (suiteConfig['provider']?.toLowerCase() === 'mongodb' && process.env.TEST_SKIP_MONGODB) continue
```

Comment:
Was not sure if moving that to `getTestSuitePlan` will make sense: here it will not even register the test suite, where test plan's skip option will still create the suite, just wraps it with `describe.skip` (otherwise jest will remove skipped snapshots). @millsp do you know if using `describe.skip` instead of bailing out will make anything worse?

Reply:
Moved this code to the `getTestSuitePlan`